### DISCUSSION TITLE
How to append attribute to results

### DISCUSSION THREAD

**Discussion by prodigy7** [spatie/laravel-query-builder#814] [Q&A]
Hi there!

I would like to add an attribute field to the queried data. I do not want to add the attribute as a default in the model via `$appends`. How would I have to adapt the following query so that the field would be appended?

```
        $tenders = \QueryBuilder::for(
            \App\Models\Tender::where('user_id', \Auth::guard('user')->user()->getKey())
                ->with('tenderAddress.geoname')
        )
            ->defaultSort('created_at')
            ->allowedSorts([
                'number_offers',
                'number_charge_points',
                'only_desired_partners',
                'created_at',
                'updated_at',
                'purpose'
            ])
            ->allowedFilters([
                'name',
            ])
            ->paginate($pageSize);
```

For example, I have already added the following before the paginate and can also access the data here, but the following paginate method then fails:
```
            ->each(function($tender) {
                //dd($tender);
                return($tender);
            })
```
It's something like a `defaultAppends` what i need

**Comment by camohub** [comment]
I also need this feature. Is there something like that? I search in IDEA hints but there is nothing like that.

**Reply by SamuelMwangiW** [reply]
I also ran into this issue. If you don't mind, I would be interested to know how you solved it. I ended up using the property to $appends which sends the attribute but now I can't restrict the allowed fields
